core/src/main/scala/cats/data/Cokleisli.scala

```diff
@@ -107,18 +107,15 @@ private trait CokleisliArrow[F[_]] extends Arrow[Cokleisli[F, ?, ?]] with Coklei
   override def dimap[A, B, C, D](fab: Cokleisli[F, A, B])(f: C => A)(g: B => D): Cokleisli[F, C, D] =
     super[CokleisliProfunctor].dimap(fab)(f)(g)
 
-  override def split[A, B, C, D](f: Cokleisli[F, A, B], g: Cokleisli[F, C, D]): Cokleisli[F, (A, C), (B, D)] =
-    super[CokleisliSplit].split(f, g)
+  override def split[A1, B1, A2, B2](f: Cokleisli[F, A1, B1], g: Cokleisli[F, A2, B2]): Cokleisli[F, (A1, A2), (B1, B2)] =
+    Cokleisli(fac => f.run(F.map(fac)(_._1)) -> g.run(F.map(fac)(_._2)))
```

Comment:
The type does imply that in each output value `(b,d): (B,D)`, to obtain the `B` (resp. `D`) you need to execute `f` (resp `g`) on the input component of type `A` (resp. `C`), but that leaves space to at least two possible implementations: one in which `f` runs before `g`, and one in which `g` runs before `f`. These implementations are not equivalent if the effects interfere, for instance, if they each one echoes and prints its input into `stdout`. That is why `split` is only really bound, apart from its definition, by commutativity. We can test this line if we write the _dual_ of the commutative monad, and use it to derive instances as done for `Kleisli`. 

The conventional implementation (Haskell) is to run `f` _first_, which I guess is because `f` refers to the _first_ components of input and output.